lib/cache/cache.go

```diff
@@ -1818,3 +1831,28 @@ func buildListResourcesResponse[T types.ResourceWithLabels](resources iter.Seq[T
 
 	return &resp, nil
 }
+
+// GetCacheSize returns the combined total number of nodes, apps, dbs, kubes, desktops, and bot instances.
+func (c *Cache) GetCacheSize() int {
```

Comment:
Should this grab the cache read lock and bail out with a -1 or an error if the cache is not ready for reading?